### PR TITLE
Adding commenting and searching functionality to CPO

### PR DESCRIPTION
**Pull request by jchen**
# Introduction
This adds editor functionality based on brownplt/pyret-codemirror-mode#10 (and depends on this PR). Added are the following keybinds:
* <kbd>Control/Cmd</kbd>-<kbd>/</kbd> to comment lines or multiple lines. 
* <kbd>Control/Cmd</kbd>-<kbd>F</kbd> and <kbd>Option</kbd>-<kbd>Control/Cmd</kbd>-<kbd>/</kbd> to find and replace respectively. 

# Testing
Tested by running locally. Also feel free to bump line 53 of `package.json`:
```
 - "pyret-codemirror-mode": "git://github.com/brownplt/pyret-codemirror-mode.git#master",
 + "pyret-codemirror-mode": "git://github.com/jchen/pyret-codemirror-mode.git#master",
```
to use Codemirror mode changes from brownplt/pyret-codemirror-mode#10 atop CPO to test new functionality.

### CODE REVIEW

**Comment by asolove** [comment]
Oh, this is cool! Please add to the shortcuts dialog as well so they are discoverable.

File: src/web/js/beforePyret.js [hunk 1]
```diff
@@ -183,6 +183,9 @@ $(function() {
       rulersMinCol = CODE_LINE_WIDTH;
     }
 
+    const mac = CodeMirror.keyMap.default === CodeMirror.keyMap.macDefault;
```
Comment (jpolitz) [inline src/web/js/beforePyret.js:186]: Oh neat, didn't know this existed.

File: src/web/js/beforePyret.js [hunk 2]
```diff
@@ -194,7 +197,8 @@ $(function() {
         "Esc Right": "goForwardSexp",
         "Alt-Right": "goForwardSexp",
         "Ctrl-Left": "goBackwardToken",
-        "Ctrl-Right": "goForwardToken"
+        "Ctrl-Right": "goForwardToken",
+        [`${modifier}-/`]: "toggleComment",
```
Comment (jpolitz) [inline src/web/js/beforePyret.js:201]: Ah OK no change for `{modifier}` here, which is least surprise. Thanks!

**Review by jpolitz — COMMENTED** [review]
All seems solid to me! I like @asolove 's comment, that's the only change I'd make. Thanks for following our very-manual makefile strategy for these files 😬

File: src/web/js/beforePyret.js (hunk 2 above)
Comment (jchen) [reply src/web/js/beforePyret.js:201]: Yup - wanted to retain the same functionality for all other commands (on a mac, Cmd-Right/Left) are still start/end of line.

**Comment by jchen** [comment]
Added all the shortcut dialogs! I didn't know this help menu was a thing so I did learn something new! 

While I was in there, I styled the help dialog a bit - hope that's reasonable: 
image

**Comment by jpolitz** [comment]
OK couple comments now that I've played with this:

I'd like to disable it in REPL entries. These can get focused for lots of reasons and it looks super weird there and completely occludes code in single-line entries 
image

Is there a way to get a graphical/clickable close button for it? Many users will not think to press escape *while focused on the search box* especially if they brought it up by accident by fat-fingering Ctrl-V

**Comment by jchen** [comment]
@jpolitz These both should be possible - I might need to tinker with it a bit. 

I suppose an unintended consequence of importing the script globally was that it applies to all CodeMirror instances which I didn't realize. 

Feel free to revert the PR if this blocks other horizon changes from getting merged.

**Comment by jpolitz** [comment]
Nah I don't think it's worth reverting! It's a good improvement for a lot of people. We should just improve it.

**Pull request merged**